### DISCUSSION TITLE
audio

### DISCUSSION THREAD

**Discussion by elbocia** [aclap-dev/video-downloadhelper#625] [Help and Q&A]
### Your question

why when i download videos it has no audio ? tried this with 3 different browsers the result was always the same. I even tried to go to the premisions to enable the audio premision but still no audio

### Addon & CoApp version and details

version: 9.0.1.18
target: google
channel: stable
lang: en-US
coapp: {"found":true,"new_version":false,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19"}
license: {"unset":true}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/125.0.0.0 Safari/537.36
{
    "download_directory": "C:\\Users\\tomas\\dwhelper",
    "theme": "system",
    "view_options": {
        "all_tabs": false,
        "hide_downloaded": false,
        "low_quality": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "sort_by_status": true,
        "sort_reverse": false
    },
    "open_count_store": 1
}

**Comment by GrampaWildWilly** [comment]
Some example URLs?  Don't give a web site.  Give a specific web page.  Don't make it hard for somebody to help you.  So, a few URLs would be greatly helping your cause.

**Reply by elbocia** [reply]
theres no specific video. just all videos i downloaded

**Reply by GrampaWildWilly** [reply]
That's fine.  Just give me something to work with.  I am not experiencing this problem so you're going to have to give me some clues.  At least one URL, maybe more if you think they're different enough & would illustrate the issue better.  Just make it a specific video, not a web site.  I need something specific to test.

**Reply by elbocia** [reply]
https://www.motogp.com/en/videos/2024/05/26/bagnaia-that-win-wasnt-just-for-the-championship-it-was-for-me/499954?playlistId=425969 well like i said it's every video i download, but i you want a specific one i picked a random one. which version of the app are you using and do you have the license ? 
![Screenshot 2024-05-27 003924](https://github.com/aclap-dev/video-downloadhelper/assets/170775603/3869967d-2c97-4339-b100-6f8fa8a07ad2)
i must be doing something wrong here a screenshot it clearly says audio & video when i hover over the downloader tab but when i download theres no audio

**Reply by GrampaWildWilly** [reply]
I do not work for VDH.  I am just another user.

I am running Windows 11 Pro 64-bit version 23H2 build 22631.3593, Firefox 126.0 64-bit, licensed VDH 9.0.1.22a1 beta, CoApp 2.0.19.

Thank you for that link.  It is useful.  I visited that page & the VDH menu was empty.  So I clicked the Play button in the player on the web page, then paused playback after a few seconds.  That made VDH recognize what you can see in this image:

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/e73d47c7-d965-415e-901d-8692b6f884d0)

I downloaded the 1080p variant shown there & these were my results:

![#02](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/348b5bc9-5406-4ec0-91ce-98e13d99ad17)

Most disturbing.  So I can confirm your experience.

But I never give up there.  Time for Plan B.  In that image, you can see an entry in the Network Monitor with the Type of dash.  (The image is kind of small.  That's a side effect of how GitHub works.  Just click the image.  It will open in a new browser window.  Then you can Ctrl-+ to magnify it & prove I speak truth.)  This is a DASH master manifest.  As with all master manifests, DASH or HLS, the next step is always to run the manifest through ffprobe.  Full disclosure.  By the time I tried to do the ffprobe, the manifest had expired.  So I had to reload the page & hit Play again.  Since I was quicker this second time, I managed to catch the manifest before it expired.  This is what was in it:

[ffprobe.txt](https://github.com/aclap-dev/video-downloadhelper/files/15449408/ffprobe.txt)

There's 5 video tracks of various resolutions there & 1 audio track.  ONE AUDIO TRACK.  Interesting.  Why didn't VDH find that?  That's a question for @paulrouget Paul.  Like I said, I'm just another dumb user.  Paul is the developer.  Anyway, using the information from that ffprobe report, I ran ffmpeg to get this video:

[Test mp4 Log.txt](https://github.com/aclap-dev/video-downloadhelper/files/15449431/Test.mp4.Log.txt)

Despite the rather unexpected error terminating the download, it did download the video, as shown by the ffprobe tacked on the end of that log.  It's a pretty short log file compared to most of the ones I get.  I suppose that's a result of a combination of the fact that it's a short video (thank you for that) & the fact that I was getting excellent download speed, which you can see in the log.  So you should read the whole log.

The file that resulted from that download looked like this:

![#03](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/b258dea6-8931-4841-ac25-eb15ec2ac277)

Hmmmmm........  There's no question now whether there's audio.  You won't be surprised to learn that it played just fine in VLC . . . WITH AUDIO.

So there is definitely something wrong with this particular video.  But you say you can't get audio with anything you download.  I say you are extraordinarily unlucky.  You must be picking sites that are all suffering from this problem.

As a test, to prove that you can get files that have audio, go here:

https://www.medici.tv/en/operas/wagner-gotterdammerung-ring-cycle-zurich-2024-gianandrea-noseda-andreas-homoki

That is a membership web site but that page is accessible to anybody without doing a login.  There is a short teaser video on that page of duration just 36 seconds.  I am able to download it with VDH no problem.  It plays fine in VLC but there doesn't seem to be any audio until about 30 seconds into the 36-second clip.  Nevertheless, there is audio there.  I want to know if you are able to download that & get audio.  If you can, then the problem is just that your favorite sites are like that biker interview.  That doesn't minimize your problem.  It just localizes it.

**Reply by elbocia** [reply]
interesting, i did download your video and it had audio i guess the problem is just in that website

**Comment by ananta-dev** [comment]
Hi @GrampaWildWilly 

Thanks for the brilliant investigation report.

I have exactly the same problem as @elbocia: Videos downloaded from https://www.motogp.com using VDH are saved without audio track (There's another website I am trying to download from with an identical issue, but I would focus on motogp.com, as it does not require logging in).

It seems you somehow managed to download that particular video (https://www.motogp.com/en/videos/2024/05/26/bagnaia-that-win-wasnt-just-for-the-championship-it-was-for-me/499954?playlistId=425969) with audio by playing with the manifest using ffprobe. 

Could the actions that allowed you to download the audio be replicated and used as a reliable workaround to download these videos until Paul (@aclap-dev) is able to handle this particular case with the extension/app?

If so, what would be the steps to take to implement that workaround?

Thank you
Juan (JJ)

**Reply by GrampaWildWilly** [reply]
Yes, the technique generalizes.  I posted a `Test mp4 Log.txt` file above.  You need to read it.  Near the top, there is an invocation of ffmpeg.  You need to study that.  That was done on May 26, 2 months ago.  There's been a few new releases of Firefox since then so you would change the `user_agent` parameter value.  You also need to learn how to use ffmpeg.  I covered that in our old forum on Google Groups:

https://groups.google.com/g/video-downloadhelper-q-and-a/c/oYDFx367MSY

We also posted many applications of that tutorial here:

https://groups.google.com/g/video-downloadhelper-q-and-a/c/sNfTCMYfiTU

You have some learning to do.

**Reply by ananta-dev** [reply]
Hi @GrampaWildWilly 
I am guessing this is the invocation you're referring to:

`C:\DownloadHelper CoApp\ffmpeg.exe" -fflags +discardcorrupt  -user_agent "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:126.0) Gecko/20100101 Firefox/126.0"  -hwaccel auto -protocol_whitelist file,crypto,data,http,https,tls,tcp -strict experimental  -analyzeduration 25000000 -probesize 250000000  -i ...
`

I could probably study this and try to figure it out (I am a software developer myself), but this seems too complex a study project when my goal is just to download several videos and I am just too busy building algorithmic trading systems in MQL5 and Python.

I was hoping for some "advanced workaround", but I understand that is not possible.

It seems the support team cannot give me a time estimate for a fix.

I guuess I will have to look for another tool.

Thank you.

**Comment by Mattmjs** [comment]
I'm guessing it would be a simple fix and the video gets downloaded with the audio. 

Until that happens I found another way, which is wait for it......  a bit of suspense, it's VLC media player. I was not 100% sure it would work
but I thought I'll try it anyway and it worked. The resulting video was there with audio. In the VDH menu use the copy url button.
Open VLC and press Ctrl + R, it might be command + R on Mac. Click on Network and paste the url then click Convert/Save.

On the next dialog for the profile it should be Video - H.264 + MP3 (MP4). If not choose this option. We are not done yet on this dialog.
Next to this profile is a spanner icon to edit profile. Click this and on Video codec make sure the Keep original video track is ticked.
Do the same for the Audio codec. Click save. After this click browse to choose a directory and your filename. Click save.
After that click Start to download the video.
